opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java

```diff
@@ -111,11 +112,11 @@ public SQLiteDatabase getWritableDatabase() {
 
     @Override
     public synchronized SQLiteDatabase getReadableDatabase(String password) {
+        if (StringUtils.isBlank(password)) {
+            throw new IllegalStateException("Password is blank");
```

Comment:
Should this return at this point?

Reply:
And in that case, would it be ok to return a null db.

^ This question applies to the writable db as well?

Reply:
The application can be able to recover from this point, so that the reason an exception is thrown. 

Returning a null is not best practice if an application can be able to recover from an exception.